rd-net/Lifetimes/Lifetimes/Lifetime.cs

```diff
@@ -83,7 +83,7 @@ public enum LifetimeStatus
   public readonly struct Lifetime : IEquatable<Lifetime>
   {        
     
-    [CanBeNull] private readonly LifetimeDefinition myDefinition;   
+    private readonly LifetimeDefinition? myDefinition;   
     [NotNull] internal LifetimeDefinition Definition => myDefinition ?? LifetimeDefinition.Eternal;   
```

Comment:
It is struct. Technically it can be null when default struct constructor is used without any warning from compiler/ReSharper (except false [one](https://youtrack.jetbrains.com/issue/RSRP-486051)). But I am not aware of any state-of-art solution for this issue.